baloise.yml

```diff
@@ -98,6 +99,56 @@ rules:
           - '511'
           - default
 
+  should-use-well-understood-http-status-codes: off
+
+  should-use-additional-well-understood-http-status-codes:
+    message: '{{error}}'
+    description: MUST use standard HTTP status codes [150]
+    documentationUrl: https://github.com/baloise-incubator/spectral-ruleset/blob/main/doc/rules/should-use-additional-well-understood-http-status-codes.md
+    severity: warn
+    given: $.paths.*
+    then:
+      function: assert-http-codes-for-operation
+      functionOptions:
+        wellUnderstood:
+          # Success Codes
+          '200': [ ALL ]
+          '201': [ POST, PUT ]
+          '202': [ POST, PUT, DELETE, PATCH ]
+          '204': [ PUT, DELETE, PATCH ]
+          '207': [ POST ]
+
+          # Redirection Codes
+          '301': [ ALL ]
+          '303': [ PATCH, POST, PUT, DELETE ]
+          '304': [ GET, HEAD ]
+
+          # Client Side Error Codes
+          '400': [ ALL ]
+          '401': [ ALL ]
+          '403': [ ALL ]
+          '404': [ ALL ]
+          '405': [ ALL ]
+          '406': [ ALL ]
+          '408': [ ALL ]
+          '409': [ POST, PUT, DELETE, PATCH ]
+          '410': [ ALL ]
+          '412': [ PUT, DELETE, PATCH ]
+          '415': [ POST, PUT, DELETE, PATCH ]
+          '422': [ ALL ]
```

Comment:
According of the definition https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 I would argue that this should only happen for POST, PUT, PATCH, as only there you should have content.

Reply:
what about query params with content?

Reply:
We can see it this way but the official docs states that 415 (Unsupported Media Type) status code is inappropriate and 400 is too, as it's a semantical error, so one could think to narrow the verbs down to at most POST, PUT, DELETE, PATCH.

I think is comes down how we want to threat content-type related status codes regarding query parameters, as they can take content-types of text or json, with out declaring the content type by header.

Would we send a 415 if we provide a query parameter named ABC with some json as value but a string was expected?

Reply:
It is not about the validity of the type / the structure, but rather about the content. 

Example: GET /locations?zipCode=413233

This is semantically correct, but from a business perspective 413233 is not a valid zipCode.
So in my understanding a 422 would be correct in this case. 

Other approach could be a 200 with an empty array as response.

Reply:
For me this is fine, just wanted to propose to narrow it down, as we can make it open in the future anyway 😉 

In you example I would definitely return 200 with an empty array, as it seam to be a search/filter query and if noting is found I would return an empty array. If we do a validation regarding zipCodes, I would rather return a 400 validation error, as most frameworks handle such errors with a 400 (e.g. spring and hibernate-validator).